contracts/dk/DuelistKingStaking.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.4 <0.9.0;
+pragma abicoder v2;
+
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/ERC20.sol';
+import '@openzeppelin/contracts/utils/Address.sol';
+import '../libraries/RegistryUser.sol';
+
+contract DuelistKingStaking is RegistryUser {
+  using SafeERC20 for ERC20;
+  using Address for address;
+
+  /**
+   * numberOfLockDays is a number of days that
+   * user must be stake before unstaking without penalty
+   */
+  struct StakingCampaign {
+    uint64 startDate;
+    uint64 endDate;
+    uint128 returnRate;
+    uint256 maxAmountOfToken;
+    uint256 stakedAmountOfToken;
+    uint256 limitStakingAmountForUser;
+    address tokenAddress;
+    uint256 maxNumberOfBoxes;
+    uint256 rewardPhaseBoxId;
+    uint64 numberOfLockDays;
+  }
+
+  struct UserStakingSlot {
+    uint256 stakingAmountOfToken;
+    uint256 stakedAmountOfBoxes;
+    uint64 startStakingDate;
+    uint64 lastStakingDate;
+  }
+
+  address private _owner;
+  uint256 private totalCampaign;
+  uint32 constant transferRate = 1000000;
+
+  mapping(uint256 => StakingCampaign) private _campaignStorage;
+
+  mapping(uint256 => mapping(address => UserStakingSlot)) private _userStakingSlot;
+
+  // New created campaign event
+  event NewCampaign(
+    uint64 indexed startDate,
+    uint64 indexed endDate,
+    uint64 numberOfLockDays,
+    uint256 maxAmountOfToken,
+    uint256 maxNumberOfBoxes,
+    address indexed tokenAddress
+  );
+
+  // Staking event
+  event Staking(address indexed owner, uint256 indexed amount, uint256 indexed startStakingDate);
+
+  // Unstaking event
+  event Unstaking(address indexed owner, uint256 indexed amount, uint256 indexed unStakeTime);
+
+  // Issue box to user evnt
+  event IssueBoxes(address indexed owner, uint256 indexed rewardPhaseBoxId, uint256 indexed numberOfBoxes);
+
+  constructor(address registry_, bytes32 domain_) {
+    _registryUserInit(registry_, domain_);
+  }
+
+  function createNewStakingCampaign(StakingCampaign memory _newCampaign) external onlyAllowSameDomain('Operator') {
```

Comment:
`Operator` is highest role in infrastructure, please use another `role` or `name`